scripts/merge-script/cli.ts

```diff
@@ -0,0 +1,183 @@
+import chalk from 'chalk';
+import {prompt} from 'inquirer';
+import * as minimist from 'minimist';
+import {isAbsolute, resolve} from 'path';
+import {Config, readAndValidateConfig} from './config';
+
+import {
+  MergeResult,
+  MergeStatus,
+  PullRequest,
+  PullRequestFailure,
+  PullRequestMergeTask,
+  PullRequestStatus
+} from './index';
+
+main();
+
+/** List of pull request statuses which are not critical and can be forcibly ignored. */
+const FORCE_OVERRIDABLE_FAILURES = [
+  PullRequestStatus.PENDING_CI_JOBS,
+  PullRequestStatus.FAILING_CI_JOBS,
+];
+
+/**
+ * Entry-point for the merge script CLI. The script can be used to merge multiple pull requests
+ * into branches based on the `PR target` labels that have been set in a configuration. The script
+ * aims to reduce the manual work that needs to be performed to cherry-pick a PR into multiple
+ * branches based on a target label.
+ */
+async function main() {
+  const {config, pullRequests, force, dryRun, githubToken: cliGithubToken} = parseCommandLine();
+  const githubToken = cliGithubToken || process.env.GITHUB_TOKEN || process.env.TOKEN;
+
+  if (!githubToken) {
+    console.error(
```

Comment:
I wonder if it would be worth creating a common util that contains `error()`, `info()` and `log()`, that perform `console.error`, `console.info`, etc. with the `chalk` colors already applied?   We are going to have to do a ton of logging things like this throughout our tooling and consistency would be nice.

If thats something that sounds like a good idea to you, I would also throw out the suggestion of these functions taking in multiple args, and placing each arg on its own console line.

It would allow us to have this block become, IMO, more readable as
```js
if (!githubToken) {
  logging.error(
    'No Github token is set. Please set the `GITHUB_TOKEN` environment variable.',
    'Alternatively, pass the `--github-token` command line flag.');
  process.exit(1);
}
```

Reply:
I like the idea of having a shared logging util. The only thing that concerns me is that usually the chalk statements are needed in the middle of a message. e.g `chalk.red(<..> ${chalk.bold(...)} <..>)`. That might be inconsistent then with the mid-sentence color changes.

Regardless of that though, I think that it would make sense to explore this. Potentially we could also have a `debug` method then (that is based on an environment variable).

Should we do that separately in the `dev-infra/` package?

Reply:
Yeah, I think its probably worth waiting and just doing it in the dev-infra package.